content.go

```diff
@@ -60,14 +63,25 @@ type ResolveOptions struct {
 }
 
 // Resolve resolves a descriptor with provided reference from the target.
-func Resolve(ctx context.Context, target ReadOnlyTarget, ref string, opts ResolveOptions) (ocispec.Descriptor, error) {
+func Resolve(ctx context.Context, target ReadOnlyTarget, reference string, opts ResolveOptions) (ocispec.Descriptor, error) {
 	if opts.TargetPlatform == nil {
-		return target.Resolve(ctx, ref)
+		return target.Resolve(ctx, reference)
+	}
+
+	proxy := cas.NewProxy(target, cas.NewMemory())
+	return resolve(ctx, target, proxy, reference, opts)
+}
+
+// resolve resolves a descriptor with provided reference from the target, with
+// specified caching.
+func resolve(ctx context.Context, target ReadOnlyTarget, proxy *cas.Proxy, reference string, opts ResolveOptions) (ocispec.Descriptor, error) {
+	if opts.TargetPlatform == nil {
```

Comment:
Do we still need to check `opts.TargetPlatform == nil` here? I noticed both function `Resolve` and function `Fetch` have already checked `TargetPlatform == nil` before calling this `resolve` function

Reply:
Right, we don't need it. 😀